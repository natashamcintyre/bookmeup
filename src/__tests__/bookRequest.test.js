import React from 'react'
import BookRequest from '../components/bookRequest'

describe('bookRequest', () => {
  it('renders without crashing', () => {
    const component = mount(<BookRequest bookID='1' />)
    expect(component).toMatchSnapshot()
  })

  it('has submit button', () => {
    const component = mount(<BookRequest bookID='1'/>)
    expect(component.exists('form#book_request')).toBe(true)
  })

  it('has a bookID', () => {
    const component = mount(<BookRequest bookID='1'/>)
    expect(component.props().bookID).toBe('1')
  })

  it('submits a request', () => {
    const requestBookMock = jest.fn()
    const component = mount(<BookRequest requestBook={requestBookMock} bookID='1'/>)
    expect(component.find('form#book_request').exists()).toBe(true)
    component.find('form#book_request').simulate('submit')
    expect(requestBookMock.mock.calls.length).toBe(1)
  })
})
