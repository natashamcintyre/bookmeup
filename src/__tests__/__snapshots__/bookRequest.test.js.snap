// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bookRequest renders without crashing 1`] = `ReactWrapper {}`;
